### PR TITLE
make collabora icons inverted

### PR DESCRIPTION
**Pull request by luka-nextcloud**
Signed-off-by: Luka Trovic <luka@nextcloud.com>

* Resolves: #680
* Target version: master 

### Summary
The color of the `share` and `more` buttons that appear on the header for office documents is now inverted according to the change of the theme color.

http://prntscr.com/1sld74l

### CODE REVIEW

**Comment by jancborchardt** [comment]
@luka-nextcloud whenever it’s a pull request which changes something visual, mind adding screenshots for quicker design review? Thank you! :)

**Review by eneiluj — COMMENTED** [review]
Code seems fine. I'll make a real life test when I'll have deployed Collabora locally :grin:.

If we decide to include the `icon-collabora` in a css file in this app, then the change could be included in this PR.

To keep track of what happened: At the moment, `icon-collabora` class is defined in this PR: https://github.com/nextcloud/server/pull/28877

**Comment by luka-nextcloud** [comment]
@eneiluj All changes are now here :)

@jancborchardt You can find out screenshots:
image
image
image

File: css/viewer.scss [hunk 1]
```diff
@@ -21,6 +21,14 @@
 	background-image: url('../img/x-office-presentation.svg');
 }
 
+.icon-collabora {
+  cursor: pointer;
```
Comment (eneiluj) [inline css/viewer.scss:25]: This should already be done with https://github.com/nextcloud/server/blob/ee987d74303cb38b864f96660cd2ee6d6552ebfd/core/css/styles.scss#L62
@jancborchardt Right?
@luka-nextcloud Is it not?

**Review by eneiluj — APPROVED** [review]
Code looks good. You could put the class in a variable for readability but that's fine like that too.

**Review by jancborchardt — APPROVED** [review]
Good design-wise, thank you!

File: css/viewer.scss [hunk 2]
```diff
@@ -21,6 +21,14 @@
...
+  opacity: 0.6;
+  &:hover {
+    opacity: 1;
+  }
+}
```
Comment (juliushaertl) [inline css/viewer.scss:30]: Should use tabs instead of spaces :wink:

File: src/view/FilesAppIntegration.js [hunk 3]
```diff
@@ -261,7 +261,8 @@ export default {
 
 	_addHeaderShareButton() {
 		if ($('header').length) {
-			const $button = $('<div id="richdocuments-sharing"><a class="icon-shared icon-white"></a></div>')
+			const isInverted = window.OCA.Theming.inverted
```
Comment (juliushaertl) [inline src/view/FilesAppIntegration.js:264]: To be on the safe side if the theming app is disabled:

```suggestion
			const isInverted = Boolean(window?.OCA?.Theming?.inverted)
```

File: src/view/FilesAppIntegration.js [hunk 4]
```diff
@@ -277,7 +278,8 @@ export default {
 		console.debug('[FilesAppIntegration] Adding header file actions')
 		OC.unregisterMenu($('#richdocuments-actions .icon-more'), $('#richdocuments-actions-menu'))
 		$('#richdocuments-actions').remove()
-		const actionsContainer = $('<div id="richdocuments-actions"><div class="icon-more icon-white"></div><ul id="richdocuments-actions-menu" class="popovermenu"></ul></div>')
+		const isInverted = window.OCA.Theming.inverted
```
Comment (juliushaertl) [inline src/view/FilesAppIntegration.js:281]: Same as https://github.com/nextcloud/richdocuments/pull/1765/files#r721170680

**Review by juliushaertl — COMMENTED** [review]
Small code style comments, otherwise looks good :+1:

**Review by juliushaertl — APPROVED** [review]

**Comment by juliushaertl** [comment]
@luka-nextcloud Quick tip for the pull request description, if you mention the issue number in the description of your pull request it is easier to check the original report and the issue will also be closed automatically once the PR gets merged, e.g. with `* Resolves: #680` as i updated the text above.

**Pull request merged**